src/diffusers/models/unets/unet_spatio_temporal_condition.py

```diff
@@ -397,6 +397,8 @@ def forward(
 
         # broadcast to batch dimension in a way that's compatible with ONNX/Core ML
         batch_size, num_frames = sample.shape[:2]
+        if torch.is_tensor(num_frames):
+            num_frames = num_frames.item()
```

Comment:
This will results in `num_frames` to be set as a constant in the ONNX graph, is this expected (I see that it's defined in the unet [config](https://huggingface.co/stabilityai/stable-video-diffusion-img2vid/blob/main/unet/config.json#L27)) or is there any cases where this value might vary @yiyixuxu @sayakpaul  ? If yes we should move the tensor to the expected device instead

Reply:
No, `num_frames` can change as it's in the pipeline call.

Reply:
@echarlaix you're right, with the suggested change `num_frames` is traced as a constant and that is undesirable. Would the below change be more suitable? 

```
        if torch.is_tensor(num_frames):
            num_frames = num_frames.to(sample.device)
```

Reply:
@echarlaix pinging to follow up on this. I ran the ONNX export of the UNET model using the above [fix](https://github.com/huggingface/diffusers/pull/6562#discussion_r1529291683). The export runs successfully. However, the model fails ONNXRuntime Inference with the error below

```
ort_s = ort.InferenceSession(model_path)
  File "/usr/local/lib/python3.10/dist-packages/onnxruntime/capi/onnxruntime_inference_collection.py", line 419, in __init__
    self._create_inference_session(providers, provider_options, disabled_optimizers)
  File "/usr/local/lib/python3.10/dist-packages/onnxruntime/capi/onnxruntime_inference_collection.py", line 452, in _create_inference_session
    sess = C.InferenceSession(session_options, self._model_path, True, self._read_config_from_model)
onnxruntime.capi.onnxruntime_pybind11_state.Fail: [ONNXRuntimeError] : 1 : FAIL : Load model from svd/svd_unet.onnx failed:Type Error: Type parameter (T) of Optype (Where) bound to different types (tensor(float) and tensor(float16) in node (/up_blocks.3/attentions.2/time_mixer/Where).
```

I'd appreciate your input on how we can move this PR along. At the moment, the ONNX export for the SpatioTemporal UNET is broken. There are 2 ways to enable the export 
1. Make `num_frames`  a scalar. This results in num_frames being an ONNX constant and not changeable during inference. 
2. Move `num_frames` to device during ONNX export. The ONNXRuntime Inference fails.

Option 1 sets `num_frames` to a Constant, but passes ONNXRuntime Inference - I'm leaning towards this as it results in immediate usability of the exported model. Happy to hear if you have alternative suggestions to the options suggested.

Reply:
> @echarlaix pinging to follow up on this. I ran the ONNX export of the UNET model using the above [fix](https://github.com/huggingface/diffusers/pull/6562#discussion_r1529291683). The export runs successfully. However, the model fails ONNXRuntime Inference with the error below

Did you try to cast `num_frames` to a different data type ? Also where you able to locate where in the graph is this issue coming from ?

Reply:
@echarlaix `num_frames` can only be an integer type as it is the `repeats` input to [repeat_interleave](https://github.com/huggingface/diffusers/blob/main/src/diffusers/models/unets/unet_spatio_temporal_condition.py#L423). int64 and int32 are the only cast options and neither resolve the issue.

Reply:
@echarlaix following up on this